homeassistant/components/gogogate2/config_flow.py

```diff
@@ -35,10 +36,12 @@ def __init__(self):
         self._ip_address = None
         self._device_type = None
 
-    async def async_step_homekit(self, discovery_info):
+    async def async_step_homekit(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> data_entry_flow.FlowResult:
         """Handle homekit discovery."""
-        await self.async_set_unique_id(discovery_info["properties"]["id"])
-        return await self._async_discovery_handler(discovery_info["host"])
+        await self.async_set_unique_id(discovery_info[zeroconf.ATTR_PROPERTIES]["id"])
```

Comment:
Should we add a constant for `id` since its a defined field for `homekit`?

Reply:
`id` is harder to pinpoint with grep, but I had a quick scan of the components and there does seem to be a few components using it. Should I add `ATTR_PROPERTIES_ID` constant to zeroconf?

Reply:
Or maybe you want me to add the constant to `homekit`?

Reply:
HomeKit discovery is a subset of zeroconf discovery. I think you have it in the right place. 

I hate to ask for a one line change, but It really should be a separate PR.

Reply:
I thought you wanted it in this PR... I have reverted it now.